src/components/refactor/dialogs/connectivity/connectivity-form.js

```diff
@@ -0,0 +1,247 @@
+import { formControlledItem } from '../../utils/form-utils';
+import TextInput from '../../inputs/text-input';
+import SelectInput from '../../inputs/select-input';
+import IntegerInput from '../../inputs/integer-input';
+import AutocompleteInput from '../../inputs/autocomplete-input';
+
+/**
+ * Hook to handle a 'connectivity value' (voltage level, bus or bus bar section)
+ * @param id optional id that has to be defined if the hook is used more than once in a form
+ * @param direction direction of placement. Either 'row' or 'column', 'row' by default.
+ * @param withPosition
+ * @param withDirectionsInfos
+ * @returns {[{voltageLevel: null, busOrBusbarSection: null},unknown]}
+ */
+export const ConnectivityForm = ({
+    id,
+    direction = 'row',
+    withDirectionsInfos = true,
+    withPosition = false,
+}) => {
+    const studyUuid = useSelector((state) => state.studyUuid);
+    const currentNode = useSelector((state) => state.currentTreeNode);
+
+    const [voltageLevelOptions, setVoltageLevelOptions] = useState([]);
+    const [busOrBusbarSectionOptions, setBusOrBusbarSectionOptions] = useState(
+        []
+    );
+
+    const [isDiagramPaneOpen, setIsDiagramPaneOpen] = useState(false);
+
+    const intl = useIntl();
+
+    const methods = useFormContext();
+    const { watch } = methods;
+
+    const {
+        id: watchVoltageLevelId,
+        topologyKind: watchVoltageLevelTopologyKind,
+    } = watch(`${CONNECTIVITY}.${VOLTAGE_LEVEL}`) || {};
+
+    useEffect(() => {
+        fetchVoltageLevels(studyUuid, currentNode?.id).then((values) => {
+            setVoltageLevelOptions(
+                values.sort((a, b) => a.id.localeCompare(b.id))
+            );
+        });
+    }, [studyUuid, currentNode?.id]);
+
+    useEffect(() => {
+        if (watchVoltageLevelId) {
+            switch (watchVoltageLevelTopologyKind) {
+                case 'NODE_BREAKER':
+                    fetchBusbarSectionsForVoltageLevel(
+                        studyUuid,
+                        currentNode?.id,
+                        watchVoltageLevelId
+                    ).then((busbarSections) => {
+                        setBusOrBusbarSectionOptions(busbarSections);
+                    });
+                    break;
+
+                case 'BUS_BREAKER':
+                    fetchBusesForVoltageLevel(
+                        studyUuid,
+                        currentNode?.id,
+                        watchVoltageLevelId
+                    ).then((buses) => setBusOrBusbarSectionOptions(buses));
+                    break;
+
+                default:
+                    setBusOrBusbarSectionOptions([]);
+                    break;
+            }
+        } else {
+            setBusOrBusbarSectionOptions([]);
+        }
+    }, [
+        watchVoltageLevelId,
+        watchVoltageLevelTopologyKind,
+        studyUuid,
+        currentNode?.id,
+    ]);
+
+    const areIdsEqual = useCallback((val1, val2) => val1.id === val2.id, []);
```

Comment:
Even if it's not a performance optimization, I like it that we understand a bit how react renders things and the meaning of the code we write, so I am in favour of keeping usecallbacks when they lead to fewer renders

Reply:
Although we could decide that we never want useCallback with a empty dependency because this means the function can be defined outside the component. However it means that if you need to modify it and use some of the state of the component later, then you need to bring the usecallback at this moment.